src/widgets/toggle-selector.ts

```diff
@@ -0,0 +1,35 @@
+import { Widget } from './widget';
+import { ElementArrayFinder } from '../wdio';
+
+export interface ToggleSelectorOption {
+    id: string;
+    caption: string;
+}
+
+export class ToggleSelector extends Widget {
+
+    public async selectOptionById(optionId: string) {
+        await this.byId(optionId).click();
+    }
+
+    public async getSelectedOption(): Promise<string> {
+        return this.byClassName('slab-option slab-selected ng-star-inserted').getText();
+    }
+
+    public async getOptions(): Promise<ToggleSelectorOption[]> {
+        const optionArray: ToggleSelectorOption[] = [];
+        const options: ElementArrayFinder = this.allByCSS("div[class*='slab-option']");
+        const numberOfOptions: number = await options.count();
+        for (let i = 0; i < numberOfOptions; i++) {
+            // We need to remove the first 2 chars because id's are being created always with 'id' prefixing the string
+            const id: string = (await options.get(i).getAttribute('id')).slice(2);
```

Comment:
Why not identifying the available options by index (0-based)? If so, the return type of this method could be just an array of strings and the index is implicit by indicating the position of the element on the array.

Reply:
If going for that approach, it would be great to have a method named "selectOptionByIndex"

Reply:
I have uploaded ToggleSelector WO's code as it was recently reshaped by @joaquimvila for the project.
Thus a brand new review is needed